clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEventProcessor.java

```diff
@@ -213,14 +217,14 @@ private void process(final SubscriptionChangeEvent ignored) {
      *              the group is sent out.
      */
     private void process(final UnsubscribeEvent event) {
-        if (!requestManagers.heartbeatRequestManager.isPresent()) {
-            KafkaException error = new KafkaException("Group membership manager not present when processing an unsubscribe event");
-            event.future().completeExceptionally(error);
-            return;
+        event.future().whenComplete((__, ___) -> subscriptions.unsubscribe());
+        if (requestManagers.heartbeatRequestManager.isPresent()) {
+            MembershipManager membershipManager = requestManagers.heartbeatRequestManager.get().membershipManager();
+            CompletableFuture<Void> future = membershipManager.leaveGroup();
+            future.whenComplete(complete(event.future()));
+        } else {
+            event.future().complete(null);
```

Comment:
This is another behavior change: `UnsubscribeEvent` can be sent even though `membershipManager` is not running 

Hence, please add comments to explain it